keps/sig-storage/20190122-csi-inline-volumes.md

```diff
@@ -110,51 +83,16 @@ spec:
-* When a volume originates from PV/PVC, the driver will receive all volume related CSI calls (normal operation).
-* When a volume originates from a pod spec (inline), the CSI driver will only receive a limited number of calls (see below).
-
-### Inline CSI lifecycle setting
-A CSI driver will have to indicate the inline volume lifecycle, that it supports, as either `persistent` or `ephemeral`.  This setting will be stored in a [`CSIDriver` configuration CRD](https://github.com/kubernetes/enhancements/issues/594) and will be checked at runtime by Kubelet and external CSI components for enforcement.  The setting is to inform the Kubelet and other CSI components how to process inline volume information for proper setup.
-
-#### Persistent inline mode
-For CSI drivers that indicate they support persistent inline volumes, the `CSIDriver.lifecycle` value should be set to `persistent` (default of if not provided).  When a driver is used to handle volumes from an inline context: 
-* Volumes are assumed to be originated from an inline pod spec.
-* The `volumeHandle` value must be generated ahead of time and manually entered in the pod spec.
-* The driver will only receive CSI calls related to storage opearations for *attach*, *mount*, *unmount*, *detach*.
-* The driver will not receive CSI calls for *porovision* and *deletion* of volumes.
-
-#### Ephemeral inline mode
-Ephemeral inline volumes are more restricted than their persistent counterparts.  The `CSIDriver.lifecycle = ephemeral` must be set to indicate support for inline ephemeral volumes which will follow the lifecycle of its pod.  For drivers indicating support for ephemral mode: 
-* Volumes are assumed to be originated from an inline pod spec.
-* The `volumeHandle` is not required and is ignored if provided in the pod spec.
-* CSI will internally generate a `volumeHandle` value which is passed to the driver.
-* Ephemeral volumes will only make CSI calls for *mount* and *unmount* volume operations.
-
-#### Omitting inline mode
-Omitting `CSIDriver.lifecycle` is an indication:
-* The driver can support PV/PVC originated volume (normal operation).
-* If the volume originates from a pod spec, it is assumed that its lifecycle is `persistent`.
+## Ephemeral inline volume proposal
+A CSI driver may be able to support either PV/PVC-originated or pod spec originated volumes. When a volume definition is embedded inside a pod spec, it is considered to be an `ephemeral inline` volume request and can only participate in *mount/unmount* volume operation calls.  Ephemeral inline volume requests have the following characteristics: 
+* The inline volume spec will not contain nor require a `volumeHandle`.
+* The CSI Kubelet plugin will internally generate a `volumeHandle` which is passed to the driver.
+* The Kubelet can send mount/unmount related calls to CSI driver that require secrets stored in Kubernetes.
```

Comment:
Note, secrets are only support during node publish, and not node unpublish.

Reply:
Will fix